### DISCUSSION TITLE
Preserve size and aspect ratio of window?

### DISCUSSION THREAD

**Discussion by emendelson** [dosemu2/dosemu2#1905] [Q&A]
When I try to move a dosemu2 text window, if I drag in the wrong place in the title bar, I stretch the window, and usually have to close and restart dosemu2 to get the original window back.

Is there an option that will prevent dosemu2 from resizing this way? (I see that there's a setting that by default prevents stretching a graphics window. Can it apply also to a text window?)

**Comment by stsp** [comment]
Yes, Ctrl-Alt-b

**Reply by stuaxo** [reply]
Ah, TIL - I guess X and Wayland aren't kind enough to let DOSEMU2 know if somebody is dragging and holding a modifier like CTRL ?

Otherwise that would be a good thing to toggle aspect ratio correctness on.

**Reply by stsp** [reply]
If dosemu doesn't see Ctrl in your
environment, then there is no point
to use dosemu at all, as Ctrl is involved
a lot in dos progs.

**Reply by stuaxo** [reply]
Ah, fair enough - for some reason I was thinking if you dragged a window frame while holding a modifier the Window manager (or whatever the wayload thing is called) wouldn't pass this through, but it was definitely just an assumption.

I can't check this as not in front of Linux box right now.

**Reply by stsp** [reply]
No, just press Ctrl-Alt-b and you can't
resize any more.

**Reply by andrewbird** [reply]
Alt + left mouse drag will move the window and you don't have to click on the border, anywhere will do.

**Comment by emendelson** [comment]
OK - thank you. I didn't know about Ctrl-Alt-B. But for users who don't know or remember this, could there be an option in the conf file that would prevent resizing entirely?

**Reply by stsp** [reply]
OK added $_X_noresize opt.

**Comment by emendelson** [comment]
Perfect. Thank you!

**Reply by emendelson** [reply]
While waiting for the update with the new no-resize option, I experimented with Ctrl-Alt-b. It seemed to have no effect. It didn't restore the original screen size, and didn't prevent further stretching. Probably I'm doing something wrong, but I can't guess what it might be.

**Reply by stsp** [reply]
Maybe you are running `dosemu -X`
but Ctrl-Alt-b is only implemented for
sdl. Or maybe its captured by your
WM.

**Reply by emendelson** [reply]
I simply start dosemu from the linux prompt. As you say, Ctrl-Alt-b may be captured by something. I installed the basics of gnome, but I run everything from the command line.

The new no-stretch version arrived (`2.0~pre9-7914-5a6eafb9d+202302172017~ubuntu22.04.1`). I have this in .dosemurc:

`$_X_noresize = (on)`

Now, the fonts don't stretch when I enlarge the window, but I can still resize the window, and, if I make it narrower than it was originally, the fonts become compressed. I know I'm asking a lot, but I think this is useful: would it be possible to completely prevent the window from being resized?

**Reply by stsp** [reply]
Try `dosemu -X` see if its any better.

**Reply by emendelson** [reply]
`dosemu -X` is worse: the window stretches in the same way it did before the new no-resize option.

**Comment by stsp** [comment]
Try this:
```
--- a/src/plugin/sdl/sdl.c
+++ b/src/plugin/sdl/sdl.c
@@ -964,6 +964,7 @@ static void SDL_change_mode(int x_res, int y_res, int w_x_res, int w_y_res)
 #endif
     SDL_ShowWindow(window);
     SDL_RaiseWindow(window);
+    SDL_SetWindowResizable(window, !config.X_noresize);
     if (config.X_fullscreen)
       render_gain_focus();
   }
```

**Reply by emendelson** [reply]
I haven't tried building dosemu2 yet, but will try to get this done on a Ubuntu system running under VMware, where I probably have most of the requirements already installed. Is there any chance you'll try it in your own builds? If not, I'll experiment in the next few days.

**Reply by stsp** [reply]
In my builds everything works already.

**Reply by emendelson** [reply]
Was able to build the downloaded code, but my version of sdl.c looks different from the code in your patch:

```
   SDL_ShowWindow(window);
    if (config.X_fullscreen) {
      SDL_RaiseWindow(window);
      render_gain_focus();
```
I added the new line after the first line shown above, and the result was exactly what I asked for: I can't resize the window. Thank you!

**Reply by stsp** [reply]
OK, added.
What version of libsdl2-dev is that?

**Reply by emendelson** [reply]
dpkg-query -s libsdl2-dev reports:

```
Package: libsdl2-dev
Status: install ok installed
Priority: optional
Section: libdevel
Installed-Size: 9442
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: same
Source: libsdl2
Version: 2.0.20+dfsg-2ubuntu1.22.04.1
``

**Reply by stsp** [reply]
What, 2.0???
I have 2.24:
```
$ dpkg-query -s libsdl2-dev
Package: libsdl2-dev
Status: install ok installed
Priority: optional
Section: libdevel
Installed-Size: 5759
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: same
Source: libsdl2
Version: 2.24.0+dfsg-1
```

Hmm, not sure, perhaps the versioning
have changed. In any case, this bug is
fixed in sdl and is worked-around also
here.

**Reply by emendelson** [reply]
Strange about the versions. This is Ubuntu Mate, if that makes any difference....
